wartremover plugin breaks scala 2.11 compilation.

wartremover plugin breaks compilation on scala 2.11 and 2.13 projects. When added it errors out with:

```
Caused by: java.lang.BootstrapMethodError: java.lang.NoClassDefFoundError: scala/runtime/java8/JFunction0$mcZ$sp
	at org.wartremover.WartTraverser.isSynthetic(WartTraverser.scala:75)
	at org.wartremover.WartTraverser.isSynthetic$(WartTraverser.scala:71)
	at org.wartremover.warts.Any$.isSynthetic(Any.scala:4)
	at org.wartremover.warts.ForbidInference$$anon$1.traverse(ForbidInference.scala:27)
	at org.wartremover.Plugin$Traverser$$anon$1.$anonfun$apply$2(Plugin.scala:74)
	at org.wartremover.Plugin$Traverser$$anon$1.$anonfun$apply$2$adapted(Plugin.scala:74)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at org.wartremover.Plugin$Traverser$$anon$1.go$1(Plugin.scala:74)
	at org.wartremover.Plugin$Traverser$$anon$1.apply(Plugin.scala:77)
	at scala.tools.nsc.Global$GlobalPhase$$anonfun$applyPhase$1.apply$mcV$sp(Global.scala:467)
	at scala.tools.nsc.Global$GlobalPhase.withCurrentUnit(Global.scala:458)
	at scala.tools.nsc.Global$GlobalPhase.applyPhase(Global.scala:467)
	at scala.tools.nsc.Global$GlobalPhase$$anonfun$run$1.apply(Global.scala:425)
	at scala.tools.nsc.Global$GlobalPhase$$anonfun$run$1.apply(Global.scala:425)
	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
	at scala.tools.nsc.Global$GlobalPhase.run(Global.scala:425)
	at scala.tools.nsc.Global$Run.compileUnitsInternal(Global.scala:1528)
	at scala.tools.nsc.Global$Run.compileUnits(Global.scala:1513)
	at scala.tools.nsc.Global$Run.compileSources(Global.scala:1508)
	at scala.tools.nsc.Global$Run.compile(Global.scala:1609)
	at xsbt.CachedCompiler0.run(CompilerInterface.scala:116)
	at xsbt.CachedCompiler0.run(CompilerInterface.scala:95)
	at xsbt.CompilerInterface.run(CompilerInterface.scala:26)
	at sbt.compiler.AnalyzingCompiler.call(AnalyzingCompiler.scala:107)
	at sbt.compiler.AnalyzingCompiler.compile(AnalyzingCompiler.scala:53)
	at sbt.compiler.AnalyzingCompiler.compile(AnalyzingCompiler.scala:47)
	at sbt.compiler.AggressiveCompile$$anonfun$3$$anonfun$compileScala$1$1.apply$mcV$sp(AggressiveCompile.scala:95)
	at sbt.compiler.AggressiveCompile$$anonfun$3$$anonfun$compileScala$1$1.apply(AggressiveCompile.scala:95)
	at sbt.compiler.AggressiveCompile$$anonfun$3$$anonfun$compileScala$1$1.apply(AggressiveCompile.scala:95)
	at sbt.compiler.AggressiveCompile.sbt$compiler$AggressiveCompile$$timed(AggressiveCompile.scala:168)
	at sbt.compiler.AggressiveCompile$$anonfun$3.compileScala$1(AggressiveCompile.scala:94)
	at sbt.compiler.AggressiveCompile$$anonfun$3.apply(AggressiveCompile.scala:142)
	at sbt.compiler.AggressiveCompile$$anonfun$3.apply(AggressiveCompile.scala:84)
	at sbt.inc.IncrementalCompile$$anonfun$doCompile$1.apply(Compile.scala:66)
	at sbt.inc.IncrementalCompile$$anonfun$doCompile$1.apply(Compile.scala:64)
	at sbt.inc.IncrementalCommon.cycle(IncrementalCommon.scala:32)
	at sbt.inc.Incremental$$anonfun$1.apply(Incremental.scala:72)
	at sbt.inc.Incremental$$anonfun$1.apply(Incremental.scala:71)
	at sbt.inc.Incremental$.manageClassfiles(Incremental.scala:99)
	at sbt.inc.Incremental$.compile(Incremental.scala:71)
	at sbt.inc.IncrementalCompile$.apply(Compile.scala:54)
	at sbt.compiler.AggressiveCompile.compile2(AggressiveCompile.scala:159)
	at sbt.compiler.AggressiveCompile.compile1(AggressiveCompile.scala:68)
	at com.typesafe.zinc.Compiler.compile(Compiler.scala:207)
	at com.typesafe.zinc.Compiler.compile(Compiler.scala:189)
	at com.typesafe.zinc.Compiler.compile(Compiler.scala:180)
	at com.typesafe.zinc.Compiler.compile(Compiler.scala:171)
	at org.gradle.api.internal.tasks.scala.ZincScalaCompiler$Compiler.execute(ZincScalaCompiler.java:86)
	at org.gradle.api.internal.tasks.scala.ZincScalaCompiler.execute(ZincScalaCompiler.java:57)
	at org.gradle.api.internal.tasks.scala.ZincScalaCompiler.execute(ZincScalaCompiler.java:40)
	at org.gradle.api.internal.tasks.compile.daemon.AbstractDaemonCompiler$CompilerWorkAction.execute(AbstractDaemonCompiler.java:113)
	at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:47)
	at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:46)
	at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:36)
	at org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:98)
	at org.gradle.workers.internal.AbstractClassLoaderWorker.executeInClassLoader(AbstractClassLoaderWorker.java:36)
	at org.gradle.workers.internal.IsolatedClassloaderWorker.execute(IsolatedClassloaderWorker.java:54)
	at org.gradle.workers.internal.WorkerDaemonServer.execute(WorkerDaemonServer.java:56)
	at org.gradle.process.internal.worker.request.WorkerAction.run(WorkerAction.java:118)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:412)
	... 3 more
Caused by: java.lang.NoClassDefFoundError: scala/runtime/java8/JFunction0$mcZ$sp
	... 68 more
Caused by: java.lang.ClassNotFoundException: scala.runtime.java8.JFunction0$mcZ$sp
	... 68 more
```

scala 2.12 seems to be working fine.

--- a/src/main/groovy/cz/augi/gradle/wartremover/WartremoverPlugin.groovy
+++ b/src/main/groovy/cz/augi/gradle/wartremover/WartremoverPlugin.groovy
@@ -30,9 +30,12 @@ class WartremoverPlugin implements Plugin<Project> {
     }
 
     private String getScalaVersion(Project p) {
-        def scalaLibrary = p.configurations.getByName('compile').dependencies.find { it.group == 'org.scala-lang' && it.name == 'scala-library' }
+        def scalaLibrary = p.configurations
+                .findAll { ['compile', 'api', 'implementation'].contains(it.name) }
+                .collectMany { it.dependencies }
+                .find { it.group == 'org.scala-lang' && it.name == 'scala-library' }
         if (!scalaLibrary) {
-            p.logger.warn('Scala library dependency not found in \'compile\' configuration, defaulting to 2.12')
+            p.logger.warn('Scala library dependency not found in \'compile\' configurations, defaulting to 2.12')
             return '2.12'
         }
         if (!isValidScalaVersion(scalaLibrary.version)) {

--- a/src/test/groovy/cz/augi/gradle/wartremover/WartremoverPluginTest.groovy
+++ b/src/test/groovy/cz/augi/gradle/wartremover/WartremoverPluginTest.groovy
@@ -24,11 +24,28 @@ class WartremoverPluginTest extends Specification {
         }
         project.evaluate()
         then:
+        assert project.configurations.getByName("wartremover").dependencies.find { it.name == 'wartremover_2.12' }
         def compileTask = project.tasks.compileScala as ScalaCompile
-        compileTask.scalaCompileOptions.additionalParameters.find { it.contains('MyProduction') }
-        !compileTask.scalaCompileOptions.additionalParameters.find { it.contains('MyTest') }
+        assert compileTask.scalaCompileOptions.additionalParameters.find { it.contains('MyProduction') }
+        assert !compileTask.scalaCompileOptions.additionalParameters.find { it.contains('MyTest') }
         def compileTestTask = project.tasks.compileTestScala as ScalaCompile
-        compileTestTask.scalaCompileOptions.additionalParameters.find { it.contains('MyProduction') }
-        compileTestTask.scalaCompileOptions.additionalParameters.find { it.contains('MyTest') }
+        assert compileTestTask.scalaCompileOptions.additionalParameters.find { it.contains('MyProduction') }
+        assert compileTestTask.scalaCompileOptions.additionalParameters.find { it.contains('MyTest') }
+    }
+
+    def "should infer compilation target correctly from different compile configurations"() {
+        def project = ProjectBuilder.builder().build()
+        when:
+        project.plugins.apply 'scala'
+        project.plugins.apply 'wartremover'
+        project.repositories {
+            jcenter()
+        }
+        project.dependencies {
+            implementation 'org.scala-lang:scala-library:2.11.12'
+        }
+        project.evaluate()
+        then:
+        assert project.configurations.getByName("wartremover").dependencies.find { it.name == 'wartremover_2.11' }
     }
 }

